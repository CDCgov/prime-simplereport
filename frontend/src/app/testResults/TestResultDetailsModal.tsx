import { gql } from "@apollo/client";
import Modal from "react-modal";
import classnames from "classnames";

import iconClose from "../../img/close.svg";
import "./TestResultPrintModal.scss";
import { QueryWrapper } from "../commonComponents/QueryWrapper";
import { TestResult } from "../testQueue/QueueItem";
import { formatFullName } from "../utils/user";
import { symptomsStringToArray } from "../utils/symptoms";
import {
  PregnancyCode,
  pregnancyMap,
} from "../../patientApp/timeOfTest/constants";
import { formatDateWithTimeOption } from "../utils/date";

type Result = {
  dateTested: string;
  result: TestResult;
  correctionStatus: TestCorrectionStatus;
  noSymptoms: boolean;
  symptoms: string;
  symptomOnset: string;
  pregnancy: PregnancyCode;
  deviceType: {
    name: string;
  };
  patient: {
    firstName: string;
    middleName: string;
    lastName: string;
    birthDate: string;
  };
  createdBy: {
    name: {
      firstName: string;
      middleName: string;
      lastName: string;
    };
  };
};

export const testResultDetailsQuery = gql`
  query getTestResultDetails($id: ID!) {
    testResult(id: $id) {
      dateTested
      result
      correctionStatus
      symptoms
      symptomOnset
      pregnancy
      deviceType {
        name
      }
      patient {
        firstName
        middleName
        lastName
        birthDate
      }
      createdBy {
        name {
          firstName
          middleName
          lastName
        }
      }
    }
  }
`;

const labelClasses = "text-bold text-no-strike text-ink";
const containerClasses =
  "width-full font-sans-md add-list-reset border-base-lighter border-2px radius-md padding-x-2 padding-y-1";
const strikeClasses = "text-base text-strike";

interface Props {
  data: { testResult: Nullable<Result> };
  testResultId: string;
  closeModal: () => void;
}

export const DetachedTestResultDetailsModal = ({ data, closeModal }: Props) => {
  const {
    dateTested,
    result,
    correctionStatus,
    symptoms,
    symptomOnset,
    pregnancy,
    deviceType,
    patient,
    createdBy,
  } = { ...data?.testResult };

  const removed = correctionStatus === "REMOVED";
  const symptomList = symptoms ? symptomsStringToArray(symptoms) : [];

  return (
    <Modal
      isOpen={true}
      style={{
        content: {
          maxHeight: "90vh",
          width: "50em",
          position: "initial",
        },
      }}
      overlayClassName="prime-modal-overlay display-flex flex-align-center flex-justify-center"
      contentLabel="Unsaved changes to current user"
      ariaHideApp={process.env.NODE_ENV !== "test"}
    >
      <div className="display-flex flex-justify">
        <h1 className="font-heading-lg margin-top-05 margin-bottom-0">
          Test details
        </h1>
        <div className="sr-time-of-test-buttons">
          <button
            className="modal__close-button"
            style={{ cursor: "pointer" }}
            onClick={closeModal}
          >
            <img className="modal__close-img" src={iconClose} alt="Close" />
          </button>
        </div>
      </div>
      <div className="border-top border-base-lighter margin-x-neg-205 margin-top-1"></div>
      <h2 className="font-sans-md margin-top-3">Patient</h2>
      <div
        className={classnames(containerClasses, "grid-row flex-row padding-0")}
      >
        <div className="grid-col padding-2 border-right-1px border-base-lighter">
          <span
            className={classnames(
              labelClasses,
              "display-block margin-bottom-1"
            )}
          >
            Name
          </span>
          <span
            className={classnames("font-sans-lg", removed && strikeClasses)}
          >
            {patient ? formatFullName(patient) : "--"}
          </span>
        </div>
        <div className="grid-col padding-2">
          <span
            className={classnames(
              labelClasses,
              "display-block margin-bottom-1"
            )}
          >
            Date of birth
          </span>
          <span
            className={classnames("font-sans-lg", removed && strikeClasses)}
          >
            {patient?.birthDate
              ? formatDateWithTimeOption(patient.birthDate)
              : "--"}
          </span>
        </div>
      </div>
      <h2 className="font-sans-md margin-top-3">Test details</h2>
      <table className={containerClasses}>
        <tbody>
          <DetailsRow
            label="SARS-CoV-2 result"
            value={result}
            removed={removed}
          />
          <DetailsRow
            label="Test date"
            value={dateTested && formatDateWithTimeOption(dateTested, true)}
            removed={removed}
          />
          <DetailsRow
            label="Device"
            value={deviceType && deviceType.name}
            removed={removed}
          />
          <DetailsRow
            label="Symptoms"
            value={
              symptomList.length > 0 ? symptomList.join(", ") : "No symptoms"
            }
            removed={removed}
          />
          <DetailsRow
            label="Symptom onset"
            value={symptomOnset && formatDateWithTimeOption(symptomOnset)}
            indent
            removed={removed}
          />
          <DetailsRow
            label="Pregnant?"
            value={pregnancy && pregnancyMap[pregnancy]}
            removed={removed}
          />
          <DetailsRow
            label="Submitted by"
            value={createdBy?.name && formatFullName(createdBy.name)}
            removed={removed}
            last
          />
        </tbody>
      </table>
    </Modal>
  );
};

const TestResultDetailsModal = (props: Omit<Props, "data">) => (
  <QueryWrapper<Props>
    query={testResultDetailsQuery}
    queryOptions={{ variables: { id: props.testResultId } }}
    Component={DetachedTestResultDetailsModal}
    componentProps={{ ...props }}
    displayLoadingIndicator={false}
  />
);

export default TestResultDetailsModal;

type StringOrFalsey = string | null | undefined | false;

const DetailsRow = ({
  label,
  value,
  indent,
  last,
  removed,
}: {
  label: string;
  value: StringOrFalsey;
  indent?: boolean;
  last?: boolean;
  removed?: boolean;
}) => {
  const tdClasses = classnames(
    "padding-y-3",
    !last && "border-bottom-1px border-base-lighter"
  );

  const labelColumnClasses = classnames(
    labelClasses,
    tdClasses,
    "width-card-lg text-left",
    indent && "font-sans-sm padding-left-2"
  );

  return (
    <tr>
      <th className={labelColumnClasses}>{label}</th>
      <td className={classnames(tdClasses, removed && strikeClasses)}>
        {value || "--"}
      </td>
    </tr>
  );
};
