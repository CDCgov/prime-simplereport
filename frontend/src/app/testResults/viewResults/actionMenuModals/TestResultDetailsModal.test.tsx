import { render, screen } from "@testing-library/react";
import ReactDOM from "react-dom";

import { DetachedTestResultDetailsModal } from "./TestResultDetailsModal";

const nonMultiplexTestResult = {
  dateTested: "2022-01-28T17:56:48.143Z",
  results: [
    {
      disease: { name: "COVID-19" as MultiplexDisease },
      testResult: "NEGATIVE" as TestResult,
      __typename: "MultiplexResult",
    },
  ],
  correctionStatus: null,
  noSymptoms: false,
  symptoms: '{"00000":"false"}',
  symptomOnset: "myOnset",
  pregnancy: null,
  surveyData: {
    pregnancy: null,
    syphilisHistory: null,
    symptoms: '{"00000":"false"}',
    symptomOnset: "myOnset",
    noSymptoms: false,
    genderOfSexualPartners: null,
  },
  deviceType: {
    name: "Fake device",
  },
  patient: {
    firstName: "First",
    middleName: "Middle",
    lastName: "Last",
    birthDate: "08/07/1990",
  },
  createdBy: {
    name: {
      firstName: "firstName",
      middleName: "middle",
      lastName: "last",
    },
  },
};

const multiplexTestResult = JSON.parse(JSON.stringify(nonMultiplexTestResult));
multiplexTestResult["results"] = [
  {
    disease: { name: "COVID-19" as MultiplexDisease },
    testResult: "POSITIVE" as TestResult,
    __typename: "MultiplexResult",
  },
  {
    disease: { name: "Flu A" as MultiplexDisease },
    testResult: "NEGATIVE" as TestResult,
    __typename: "MultiplexResult",
  },
  {
    disease: { name: "Flu B" as MultiplexDisease },
    testResult: "POSITIVE" as TestResult,
    __typename: "MultiplexResult",
  },
  {
    disease: { name: "RSV" as MultiplexDisease },
    testResult: "UNDETERMINED" as TestResult,
    __typename: "MultiplexResult",
  },
];

const hivTestResult = JSON.parse(JSON.stringify(nonMultiplexTestResult));
hivTestResult.results = [
  {
    disease: { name: "HIV" as MultiplexDisease },
    testResult: "POSITIVE" as TestResult,
    __typename: "MultiplexResult",
  },
];
hivTestResult.surveyData.genderOfSexualPartners = ["female", "male", "other"];
hivTestResult.surveyData.pregnancy = "77386006";

const syphilisTestResult = JSON.parse(JSON.stringify(nonMultiplexTestResult));
syphilisTestResult.results = [
  {
    disease: { name: "Syphilis" as MultiplexDisease },
    testResult: "POSITIVE" as TestResult,
    __typename: "MultiplexResult",
  },
];
syphilisTestResult.surveyData.pregnancy = "77386006";
syphilisTestResult.surveyData.syphilisHistory = "1087151000119108";
syphilisTestResult.surveyData.symptoms =
  '{"91554004":"false","15188001":"false","724386005":"false","56940005":"true","266128007":"true","46636008":"true","68225006":"false","195469007":"false","26284000":"false"}';
syphilisTestResult.surveyData.symptomOnset = "2024-09-24";
syphilisTestResult.surveyData.noSymptoms = false;
syphilisTestResult.surveyData.genderOfSexualPartners = [
  "female",
  "male",
  "other",
];

const renderComponent = (testResult: TestResult) =>
  render(
    <DetachedTestResultDetailsModal
      data={{ testResult: testResult }}
      closeModal={() => {}}
    />
  );
describe("single disease TestResultDetailsModal", () => {
  beforeEach(() => {
    ReactDOM.createPortal = jest.fn((element, _node) => {
      return element;
    }) as any;
  });

  it("should render the test date and test time", () => {
    renderComponent(nonMultiplexTestResult);
    expect(screen.getByText("01/28/2022 5:56pm")).toBeInTheDocument();
  });

  it("shouldn't have flu A or B result rows", () => {
    renderComponent(nonMultiplexTestResult);
    expect(screen.queryByText("Flu A result")).not.toBeInTheDocument();
    expect(screen.queryByText("Flu B result")).not.toBeInTheDocument();
  });

  it("matches screenshot", () => {
    let view = renderComponent(nonMultiplexTestResult);
    expect(view).toMatchSnapshot();
  });
});

describe("multiple diseases TestResultDetailsModal", () => {
  beforeEach(() => {
    ReactDOM.createPortal = jest.fn((element, _node) => {
      return element;
    }) as any;
  });

  it("should render the test date and test time", () => {
    renderComponent(multiplexTestResult);
    expect(screen.getByText("01/28/2022 5:56pm")).toBeInTheDocument();
  });

  it("should have rows for all diseases in result", () => {
    renderComponent(multiplexTestResult);
    expect(screen.getByText("COVID-19 result")).toBeInTheDocument();
    expect(screen.getByText("Flu A result")).toBeInTheDocument();
    expect(screen.getByText("Flu B result")).toBeInTheDocument();
    expect(screen.getByText("RSV result")).toBeInTheDocument();
    expect(
      screen.queryByText("Gender of sexual partners")
    ).not.toBeInTheDocument();
  });

  it("matches screenshot", () => {
    let view = renderComponent(multiplexTestResult);
    expect(view).toMatchSnapshot();
  });
});

describe("HIV TestResultDetailsModal", () => {
  beforeEach(() => {
    ReactDOM.createPortal = jest.fn((element, _node) => {
      return element;
    }) as any;
  });

  it("should have HIV specific AOE text", () => {
    renderComponent(hivTestResult);
    expect(screen.getByText("HIV result")).toBeInTheDocument();
    expect(screen.queryByText("Symptoms")).not.toBeInTheDocument();
    expect(screen.queryByText("Symptom onset")).not.toBeInTheDocument();
    expect(screen.getByText("Gender of sexual partners")).toBeInTheDocument();
    expect(screen.getByText("Female, Male, Other")).toBeInTheDocument();
    expect(screen.getByText("Pregnant?")).toBeInTheDocument();
  });

  it("matches screenshot", () => {
    let view = renderComponent(hivTestResult);
    expect(view).toMatchSnapshot();
  });
});

describe("Syphilis TestResultDetailsModal", () => {
  beforeEach(() => {
    ReactDOM.createPortal = jest.fn((element, _node) => {
      return element;
    }) as any;
  });

  it("should have Syphilis specific symptoms", () => {
    renderComponent(syphilisTestResult);
    expect(screen.getByText("Syphilis result")).toBeInTheDocument();
    expect(screen.getByText("Symptoms")).toBeInTheDocument();
    expect(screen.getByText("Symptom onset")).toBeInTheDocument();
    expect(
      screen.getByText(
        "Blurred vision, Palmar (hand)/plantar (foot) rash, Body Rash"
      )
    ).toBeInTheDocument();
    expect(screen.getByText("Gender of sexual partners")).toBeInTheDocument();
    expect(screen.getByText("Female, Male, Other")).toBeInTheDocument();
    expect(screen.getByText("Pregnant?")).toBeInTheDocument();
    expect(
      screen.getByText("Previously told they have syphilis?")
    ).toBeInTheDocument();
  });

  it("matches screenshot", () => {
    let view = renderComponent(syphilisTestResult);
    expect(view).toMatchSnapshot();
  });
});
