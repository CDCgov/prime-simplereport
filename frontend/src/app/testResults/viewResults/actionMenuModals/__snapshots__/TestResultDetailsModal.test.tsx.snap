// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multiple diseases TestResultDetailsModal matches screenshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="display-flex flex-justify"
      >
        <h1
          class="font-heading-lg margin-top-05 margin-bottom-0"
          id="result-detail-title"
        >
          Result details
        </h1>
        <div
          class="sr-time-of-test-buttons"
        >
          <button
            class="modal__close-button"
            style="cursor: pointer;"
          >
            <img
              alt="Close"
              class="modal__close-img"
              src="close.svg"
            />
          </button>
        </div>
      </div>
      <div
        class="border-top border-base-lighter margin-x-neg-205 margin-top-1"
      />
      <h2
        class="font-sans-md margin-top-3"
      >
        Patient
      </h2>
      <div
        class="width-full font-sans-md add-list-reset border-base-lighter border-2px radius-md padding-x-2 padding-y-1 grid-row flex-row padding-0"
      >
        <div
          class="grid-col padding-2 border-right-1px border-base-lighter"
        >
          <span
            class="text-bold text-no-strike text-ink display-block margin-bottom-1"
          >
            Name
          </span>
          <span
            aria-describedby="result-detail-title"
            class="font-sans-lg"
          >
            First Middle Last
          </span>
        </div>
        <div
          class="grid-col padding-2"
        >
          <span
            class="text-bold text-no-strike text-ink display-block margin-bottom-1"
          >
            Date of birth
          </span>
          <span
            aria-describedby="result-detail-title"
            class="font-sans-lg"
          >
            08/07/1990
          </span>
        </div>
      </div>
      <h2
        class="font-sans-md margin-top-3"
      >
        Test information
      </h2>
      <table
        class="width-full font-sans-md add-list-reset border-base-lighter border-2px radius-md padding-x-2 padding-y-1"
      >
        <tbody>
          <tr>
            <th
              class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
            >
              COVID-19 result
            </th>
            <td
              class="padding-y-3 border-bottom-1px border-base-lighter"
            >
              POSITIVE
            </td>
          </tr>
          <tr>
            <th
              class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
            >
              Flu A result
            </th>
            <td
              class="padding-y-3 border-bottom-1px border-base-lighter"
            >
              NEGATIVE
            </td>
          </tr>
          <tr>
            <th
              class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
            >
              Flu B result
            </th>
            <td
              class="padding-y-3 border-bottom-1px border-base-lighter"
            >
              POSITIVE
            </td>
          </tr>
          <tr>
            <th
              class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
            >
              RSV result
            </th>
            <td
              class="padding-y-3 border-bottom-1px border-base-lighter"
            >
              UNDETERMINED
            </td>
          </tr>
          <tr>
            <th
              class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
            >
              Test date
            </th>
            <td
              class="padding-y-3 border-bottom-1px border-base-lighter"
            >
              01/28/2022 5:56pm
            </td>
          </tr>
          <tr>
            <th
              class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
            >
              Device
            </th>
            <td
              class="padding-y-3 border-bottom-1px border-base-lighter"
            >
              Fake device
            </td>
          </tr>
          <tr>
            <th
              class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
            >
              Symptoms
            </th>
            <td
              class="padding-y-3 border-bottom-1px border-base-lighter"
            >
              No symptoms
            </td>
          </tr>
          <tr>
            <th
              class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
            >
              Symptom onset
            </th>
            <td
              class="padding-y-3 border-bottom-1px border-base-lighter"
            >
              Invalid date
            </td>
          </tr>
          <tr>
            <th
              class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
            >
              Pregnant?
            </th>
            <td
              class="padding-y-3 border-bottom-1px border-base-lighter"
            >
              --
            </td>
          </tr>
          <tr>
            <th
              class="text-bold text-no-strike text-ink padding-y-3 width-card-lg text-left"
            >
              Submitted by
            </th>
            <td
              class="padding-y-3"
            >
              firstName middle last
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <div
      class="display-flex flex-justify"
    >
      <h1
        class="font-heading-lg margin-top-05 margin-bottom-0"
        id="result-detail-title"
      >
        Result details
      </h1>
      <div
        class="sr-time-of-test-buttons"
      >
        <button
          class="modal__close-button"
          style="cursor: pointer;"
        >
          <img
            alt="Close"
            class="modal__close-img"
            src="close.svg"
          />
        </button>
      </div>
    </div>
    <div
      class="border-top border-base-lighter margin-x-neg-205 margin-top-1"
    />
    <h2
      class="font-sans-md margin-top-3"
    >
      Patient
    </h2>
    <div
      class="width-full font-sans-md add-list-reset border-base-lighter border-2px radius-md padding-x-2 padding-y-1 grid-row flex-row padding-0"
    >
      <div
        class="grid-col padding-2 border-right-1px border-base-lighter"
      >
        <span
          class="text-bold text-no-strike text-ink display-block margin-bottom-1"
        >
          Name
        </span>
        <span
          aria-describedby="result-detail-title"
          class="font-sans-lg"
        >
          First Middle Last
        </span>
      </div>
      <div
        class="grid-col padding-2"
      >
        <span
          class="text-bold text-no-strike text-ink display-block margin-bottom-1"
        >
          Date of birth
        </span>
        <span
          aria-describedby="result-detail-title"
          class="font-sans-lg"
        >
          08/07/1990
        </span>
      </div>
    </div>
    <h2
      class="font-sans-md margin-top-3"
    >
      Test information
    </h2>
    <table
      class="width-full font-sans-md add-list-reset border-base-lighter border-2px radius-md padding-x-2 padding-y-1"
    >
      <tbody>
        <tr>
          <th
            class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
          >
            COVID-19 result
          </th>
          <td
            class="padding-y-3 border-bottom-1px border-base-lighter"
          >
            POSITIVE
          </td>
        </tr>
        <tr>
          <th
            class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
          >
            Flu A result
          </th>
          <td
            class="padding-y-3 border-bottom-1px border-base-lighter"
          >
            NEGATIVE
          </td>
        </tr>
        <tr>
          <th
            class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
          >
            Flu B result
          </th>
          <td
            class="padding-y-3 border-bottom-1px border-base-lighter"
          >
            POSITIVE
          </td>
        </tr>
        <tr>
          <th
            class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
          >
            RSV result
          </th>
          <td
            class="padding-y-3 border-bottom-1px border-base-lighter"
          >
            UNDETERMINED
          </td>
        </tr>
        <tr>
          <th
            class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
          >
            Test date
          </th>
          <td
            class="padding-y-3 border-bottom-1px border-base-lighter"
          >
            01/28/2022 5:56pm
          </td>
        </tr>
        <tr>
          <th
            class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
          >
            Device
          </th>
          <td
            class="padding-y-3 border-bottom-1px border-base-lighter"
          >
            Fake device
          </td>
        </tr>
        <tr>
          <th
            class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
          >
            Symptoms
          </th>
          <td
            class="padding-y-3 border-bottom-1px border-base-lighter"
          >
            No symptoms
          </td>
        </tr>
        <tr>
          <th
            class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
          >
            Symptom onset
          </th>
          <td
            class="padding-y-3 border-bottom-1px border-base-lighter"
          >
            Invalid date
          </td>
        </tr>
        <tr>
          <th
            class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
          >
            Pregnant?
          </th>
          <td
            class="padding-y-3 border-bottom-1px border-base-lighter"
          >
            --
          </td>
        </tr>
        <tr>
          <th
            class="text-bold text-no-strike text-ink padding-y-3 width-card-lg text-left"
          >
            Submitted by
          </th>
          <td
            class="padding-y-3"
          >
            firstName middle last
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`single disease TestResultDetailsModal matches screenshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="display-flex flex-justify"
      >
        <h1
          class="font-heading-lg margin-top-05 margin-bottom-0"
          id="result-detail-title"
        >
          Result details
        </h1>
        <div
          class="sr-time-of-test-buttons"
        >
          <button
            class="modal__close-button"
            style="cursor: pointer;"
          >
            <img
              alt="Close"
              class="modal__close-img"
              src="close.svg"
            />
          </button>
        </div>
      </div>
      <div
        class="border-top border-base-lighter margin-x-neg-205 margin-top-1"
      />
      <h2
        class="font-sans-md margin-top-3"
      >
        Patient
      </h2>
      <div
        class="width-full font-sans-md add-list-reset border-base-lighter border-2px radius-md padding-x-2 padding-y-1 grid-row flex-row padding-0"
      >
        <div
          class="grid-col padding-2 border-right-1px border-base-lighter"
        >
          <span
            class="text-bold text-no-strike text-ink display-block margin-bottom-1"
          >
            Name
          </span>
          <span
            aria-describedby="result-detail-title"
            class="font-sans-lg"
          >
            First Middle Last
          </span>
        </div>
        <div
          class="grid-col padding-2"
        >
          <span
            class="text-bold text-no-strike text-ink display-block margin-bottom-1"
          >
            Date of birth
          </span>
          <span
            aria-describedby="result-detail-title"
            class="font-sans-lg"
          >
            08/07/1990
          </span>
        </div>
      </div>
      <h2
        class="font-sans-md margin-top-3"
      >
        Test information
      </h2>
      <table
        class="width-full font-sans-md add-list-reset border-base-lighter border-2px radius-md padding-x-2 padding-y-1"
      >
        <tbody>
          <tr>
            <th
              class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
            >
              COVID-19 result
            </th>
            <td
              class="padding-y-3 border-bottom-1px border-base-lighter"
            >
              NEGATIVE
            </td>
          </tr>
          <tr>
            <th
              class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
            >
              Test date
            </th>
            <td
              class="padding-y-3 border-bottom-1px border-base-lighter"
            >
              01/28/2022 5:56pm
            </td>
          </tr>
          <tr>
            <th
              class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
            >
              Device
            </th>
            <td
              class="padding-y-3 border-bottom-1px border-base-lighter"
            >
              Fake device
            </td>
          </tr>
          <tr>
            <th
              class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
            >
              Symptoms
            </th>
            <td
              class="padding-y-3 border-bottom-1px border-base-lighter"
            >
              No symptoms
            </td>
          </tr>
          <tr>
            <th
              class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
            >
              Symptom onset
            </th>
            <td
              class="padding-y-3 border-bottom-1px border-base-lighter"
            >
              Invalid date
            </td>
          </tr>
          <tr>
            <th
              class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
            >
              Pregnant?
            </th>
            <td
              class="padding-y-3 border-bottom-1px border-base-lighter"
            >
              --
            </td>
          </tr>
          <tr>
            <th
              class="text-bold text-no-strike text-ink padding-y-3 width-card-lg text-left"
            >
              Submitted by
            </th>
            <td
              class="padding-y-3"
            >
              firstName middle last
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <div
      class="display-flex flex-justify"
    >
      <h1
        class="font-heading-lg margin-top-05 margin-bottom-0"
        id="result-detail-title"
      >
        Result details
      </h1>
      <div
        class="sr-time-of-test-buttons"
      >
        <button
          class="modal__close-button"
          style="cursor: pointer;"
        >
          <img
            alt="Close"
            class="modal__close-img"
            src="close.svg"
          />
        </button>
      </div>
    </div>
    <div
      class="border-top border-base-lighter margin-x-neg-205 margin-top-1"
    />
    <h2
      class="font-sans-md margin-top-3"
    >
      Patient
    </h2>
    <div
      class="width-full font-sans-md add-list-reset border-base-lighter border-2px radius-md padding-x-2 padding-y-1 grid-row flex-row padding-0"
    >
      <div
        class="grid-col padding-2 border-right-1px border-base-lighter"
      >
        <span
          class="text-bold text-no-strike text-ink display-block margin-bottom-1"
        >
          Name
        </span>
        <span
          aria-describedby="result-detail-title"
          class="font-sans-lg"
        >
          First Middle Last
        </span>
      </div>
      <div
        class="grid-col padding-2"
      >
        <span
          class="text-bold text-no-strike text-ink display-block margin-bottom-1"
        >
          Date of birth
        </span>
        <span
          aria-describedby="result-detail-title"
          class="font-sans-lg"
        >
          08/07/1990
        </span>
      </div>
    </div>
    <h2
      class="font-sans-md margin-top-3"
    >
      Test information
    </h2>
    <table
      class="width-full font-sans-md add-list-reset border-base-lighter border-2px radius-md padding-x-2 padding-y-1"
    >
      <tbody>
        <tr>
          <th
            class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
          >
            COVID-19 result
          </th>
          <td
            class="padding-y-3 border-bottom-1px border-base-lighter"
          >
            NEGATIVE
          </td>
        </tr>
        <tr>
          <th
            class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
          >
            Test date
          </th>
          <td
            class="padding-y-3 border-bottom-1px border-base-lighter"
          >
            01/28/2022 5:56pm
          </td>
        </tr>
        <tr>
          <th
            class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
          >
            Device
          </th>
          <td
            class="padding-y-3 border-bottom-1px border-base-lighter"
          >
            Fake device
          </td>
        </tr>
        <tr>
          <th
            class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
          >
            Symptoms
          </th>
          <td
            class="padding-y-3 border-bottom-1px border-base-lighter"
          >
            No symptoms
          </td>
        </tr>
        <tr>
          <th
            class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
          >
            Symptom onset
          </th>
          <td
            class="padding-y-3 border-bottom-1px border-base-lighter"
          >
            Invalid date
          </td>
        </tr>
        <tr>
          <th
            class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
          >
            Pregnant?
          </th>
          <td
            class="padding-y-3 border-bottom-1px border-base-lighter"
          >
            --
          </td>
        </tr>
        <tr>
          <th
            class="text-bold text-no-strike text-ink padding-y-3 width-card-lg text-left"
          >
            Submitted by
          </th>
          <td
            class="padding-y-3"
          >
            firstName middle last
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
